index.bs

```diff
@@ -987,6 +987,8 @@ these steps.
      1. Run the following steps [=in parallel=]:
          1. Gather the stats from the [=underlying connection=], including stats on datagrams.
          1. [=Queue a network task=] with |transport| to run the following steps:
+           1. If |transport|.{{[[State]]}} is `"closed"` or `"failed"`,
+              [=reject=] |p| with an {{InvalidStateError}} and abort these steps.
```

Comment:
I think we should do the same state check in the synchronous steps (since the timing of promise state is synchronously observable in a JS debugger).